moped-editor/src/queries/components.js

```diff
@@ -166,6 +175,7 @@ export const UPDATE_COMPONENT_ATTRIBUTES = gql`
     $phaseId: Int
     $subphaseId: Int
     $completionDate: timestamptz
+    $componentTags: [moped_proj_component_tags_insert_input!]!
   ) {
     update_moped_proj_components_subcomponents(
       where: { project_component_id: { _eq: $projectComponentId } }
```

Comment:
Just realized what was going on here. This update mutation is first clearing all of the subcomponents by switching any `moped_proj_components_subcomponents` associated with the current component by switch them all to `is_deleted = true`. After that update, we insert the subcomponents chosen in the form, and, if there is a conflict, we update `is_deleted = true` to `is_deleted = false`.  So, the same update for tags is needed here to match up with what happens to subcomponents.

That took me a second to figure out again so let me know if that makes sense. More comments around this with wording from other people than me would make this clearer in the future.

Reply:
I think that your comment is much clearer! Thanks for updating it. 🙏